sdk/cosmos/azure-cosmos/azure/cosmos/partition_key.py

```diff
@@ -190,31 +192,34 @@ def _get_effective_partition_key_string(self, pk_value: list):
             version = self.version or 2
             if version == 1:
                 return self._get_effective_partition_key_for_hash_partitioning()
-            elif version == 2:
+            if version == 2:
                 return self._get_effective_partition_key_for_hash_partitioning_v2(pk_value)
         elif kind == 'MultiHash':
             return self._get_effective_partition_key_for_multi_hash_partitioning_v2(pk_value)
-        else:
-            return _to_hex_encoded_binary_string(pk_value)
+        return _to_hex_encoded_binary_string(pk_value)
 
-    def _write_for_hashing_v2(self, value, writer):
+    def _write_for_hashing_v2(self, value: Any, writer: BytesIO) -> None:
         if value is True:
             writer.write(bytes([PartitionKeyComponentType.PTrue]))
         elif value is False:
             writer.write(bytes([PartitionKeyComponentType.PFalse]))
         elif value is None or value == {} or isinstance(value, NonePartitionKeyValue):
             writer.write(bytes([PartitionKeyComponentType.Null]))
-        elif isinstance(value, (int, float)):
+        elif isinstance(value, int):
             writer.write(bytes([PartitionKeyComponentType.Number]))
             writer.write(value.to_bytes(8, 'little'))  # assuming value is a 64-bit integer
+        elif isinstance(value, float):
+            # Encoding as 4-byte le float. TODO: check correct encoding.
+            writer.write(bytes([PartitionKeyComponentType.Float]))
+            writer.write(struct.pack('<f', value))
```

Comment:
I have no idea if this is the correct encoding for a float partition key - so this will need service team confirmation.
I tried to compare to .NET/Java SDKs, but couldn't find it.